_posts/2021-11-12-github-copilot.md

```diff
@@ -0,0 +1,128 @@
+---
+layout: post
+title: Github Copilot - let AI write your code?
+author: arto
+excerpt: >
+  Using tools to increase effectiveness has been a long-standing trend in the software development community. Latest wave comes in form of GPT-3, OpenAI, and ultimately Github Copilot. Let's try to answer the question: Can AI write your code?
+tags:
+ - Software Development
+ - AI
+ - GPT-3
+ - OpenAI
+ - Github Copilot
+---
+
+_One day soon, AI will be writing the code. I, for one, welcome our new robot overlords._
+
+![Copilot helping with Markdown](/img/github-copilot/copilot_helping_with_markdown_3.png)
+*Copilot is writing this blog text for me*
+
+## What is Github Copilot?
+
+Right now it's an intellisense-kind of tool you can get in popular IDEs. Whereas traditional code suggestion plugins typically detect patterns in your code and suggest code snippets, Github Copilot is reading your code and trying to understand your context, comments, and method names, and then synthesizing code suggestions. Raw material for the suggestions comes from Github public repositories. But it's not just a copy-paste snippet.
+
+You might have seen GPT-3 based code generation demos before. Copilot is powered by OpenAI Codex, which is a new AI system, more capable than GPT-3. It's currently available as a technical preview, on request. It's expected that later on it will be released as a commercial offering, open for everyone.
+
+My interest in this tool comes from always wanting to improve efficiency and learning, so I applied for technical preview, and got accepted, and have now used this for a few weeks for various personal projects.
+
+## What can Github Copilot do for you?
+
+Well, quite a lot. Once you have installed the plugin and the Copilot has been enabled for you, you can start writing code, and it will be automatically filled in with code suggestions. You can ignore them and keep on coding as usual, or press the TAB key to accept the suggestions. You can also press Ctrl+Enter to get up to 10 different suggestions for the problem.
```

Comment:
> and it will be automatically filled in with code suggestions.

Not clear what "it" is. Maybe "and Copilot will automatically fill in code suggestions."